source/slang/slang-emit-spirv.cpp

```diff
@@ -3822,22 +3821,35 @@ struct SPIRVEmitContext
                 return nullptr;
             }
 
-            const auto parentForOpCode = [this](SpvOp opcode, SpvInstParent* defaultParent){
-                return
-                    opcode == SpvOpConstant ? getSection(SpvLogicalSectionID::ConstantsAndTypes)
-                    : opcode == SpvOpName ? getSection(SpvLogicalSectionID::DebugNames)
-                    : defaultParent;
+            const auto parentForOpCode = [this](SpvOp opcode, SpvInstParent* defaultParent) -> SpvInstParent*{
+                const auto info = m_grammarInfo->opInfos.lookup(opcode);
+                SLANG_ASSERT(info.has_value());
+                switch(info->class_)
+                {
+                    case SPIRVCoreGrammarInfo::OpInfo::TypeDeclaration:
+                    case SPIRVCoreGrammarInfo::OpInfo::ConstantCreation:
```

Comment:
Can we add cases for `OpCapability` and `OpExtension` here?

Reply:
yeah, good idea, I'll do that for this PR or the next one if this merges before I get to it